File: service24/resources/utils/MathUtils.java
package utils;

import unibo.basicomm23.utils.CommUtils;

public  class MathUtils {
protected static int count = 1;
protected static int thcounter = 0;
protected static int numOfExecutors = 0;
private static MathUtils singleton;

    public static MathUtils create() {
    	if( singleton == null ) singleton = new MathUtils();
    	return singleton;
    }
    
	public  int fibo(int n) {
		if( n==0 || n==1) return 1;
		return fibo(n-1) + fibo(n-2);	
	}
 }

 

File: servicemath24Synch/src/main/kotlin/MathUtils.kt
import alice.tuprolog.Prolog
import alice.tuprolog.Theory
import it.unibo.kactor.sysUtil
import unibo.basicomm23.utils.CommUtils
import java.io.FileInputStream
import java.io.FileOutputStream
import java.util.HashMap

class MathUtils {

    companion object {
        protected var count = 1
        protected var thcounter = 0
        protected var numOfExecutors = 0
        private val map = HashMap<Int, Int>()

        fun create(): MathUtils {
         	return MathUtils()
        }
    }

    fun fibo(n: Int): Long {
        return if (n == 1 || n == 2) 1 else fibo(n - 1) + fibo(n - 2)
    }

 }//MathUtils
 